provider/datagen/src/transform/cldr/units/mod.rs

```diff
@@ -40,101 +90,94 @@ impl DataProvider<UnitsConstantsV1Marker> for crate::DatagenProvider {
-                // and replace any strings that are keys in the constants map with their corresponding values.
-                for den_str in den {
-                    if !contains_alphabetic_chars(den_str) || is_scientific_number(den_str) {
-                        temp_den.push(den_str.clone());
-                        continue;
-                    }
-                    has_internal_constants = true;
-
-                    if let Some((rnum, rden, rconstant_exactness)) =
-                        constants_map_in_str_form.get(den_str.as_str())
-                    {
-                        temp_num.extend(rden.clone());
-                        temp_den.extend(rnum.clone());
-
-                        if rconstant_exactness == &ConstantExactness::Approximate {
-                            temp_constant_exactness = *rconstant_exactness;
-                        }
+            for _ in 0..non_scientific_constant.non_scientific_den.len() {
+                if let Some(den) = non_scientific_constant.non_scientific_den.pop_front() {
+                    if let Some(clean_constant) = clean_constants_map.get(den.as_str()) {
+                        non_scientific_constant
+                            .clean_num
+                            .extend(clean_constant.clean_den.clone());
+                        non_scientific_constant
+                            .clean_den
+                            .extend(clean_constant.clean_num.clone());
+
+                        updated = true;
                     } else {
-                        temp_den.push(den_str.clone());
+                        non_scientific_constant.non_scientific_den.push_back(den);
                     }
                 }
-
-                constants_with_constants_map_replaceable
-                    .insert(cons_name, (temp_num, temp_den, temp_constant_exactness));
             }
 
-            constants_map_in_str_form = constants_with_constants_map_replaceable;
-
-            if !has_internal_constants {
-                break;
+            if non_scientific_constant.is_free_of_non_scientific() {
+                clean_constants_map.insert(constant_key, non_scientific_constant);
+            } else {
+                constants_with_non_scientific.push_back((constant_key, non_scientific_constant));
             }
-        }
 
-        if max_depth_reached >= maximum_depth {
-            return Err(DataError::custom(
-                "Maximum depth reached while parsing constants. \
-                This is likely due to a circular dependency in the constants. \
-                Note: If the depth was increased, you may need to increase the maximum depth in the code.",
-            ));
+            count = if !updated { count + 1 } else { 0 };
+            if count > constants_with_non_scientific.len() {
+                return Err(DataError::custom(
+                    "An Infinite loop was detected in the CLDR constants data!",
```

Comment:
nit: 

```suggestion
                    "A loop was detected in the CLDR constants data!",
```

Reply:
I prefer to use 'Infinite' because it clearly refers to a situation that can be infinite, and the term 'infinite loop' is well-recognized.

Reply:
"infinite loop" is a programming term for a control flow that does not terminate. The loop we're talking about here is a loop in a graph ("in the data"), where the term "infinite" doesn't really make sense. But whatever